corehq/ex-submodules/pillowtop/index_settings.py

```diff
@@ -0,0 +1,25 @@
+from corehq.elastic import SIZE_LIMIT
+
+INDEX_REINDEX_SETTINGS = {
+    "index": {
+        "refresh_interval": "1800s",
+        # this property will be filtered out on ES 1
+        "max_result_window": SIZE_LIMIT,
+        # todo: remove these deprecated properties we are off ES 1
+        "merge.policy.merge_factor": 20,
+        "store.throttle.max_bytes_per_sec": "1mb",
+        "store.throttle.type": "merge",
+    }
+}
+INDEX_STANDARD_SETTINGS = {
+    "index": {
+        "refresh_interval": "5s",
+        # this property will be filtered out on ES 1
+        "max_result_window": SIZE_LIMIT,
+        # todo: remove these deprecated properties we are off ES 1
+        "merge.policy.merge_factor": 10,
+        "store.throttle.max_bytes_per_sec": "5mb",
+        "store.throttle.type": "node",
+
+    }
+}
```

Comment:
Yeah, in a different incarnation of this I had split out these settings right here, but that didn't feel fantastic either because all the splitting logic is supposed to happen in this `corehq.util.es` package.

Another option would be to move these settings themselves into `corehq.util.es`, but then that felt too much like I was moving more than just the compatibility layer into that, and if I'm going to be introducing yet another layer to our ES system, I'd like for it to at least be small and have well-defined responsibilities—i.e. stick to just compatibility.

So then even though it's a little not DRY, I decided that here we'd put all the settings here, and then in `corehq.util.es.interface` we'd list out which settings are incompatible with which versions, and then also have comments to that effect here. Open to suggestions, but I think this might be good-enough.

Reply:
Oh interesting - is your end goal for `corehq.util.es` to focus strictly on compatibility?  I was imagining it might eventually replace `corehq.elastic` and hide all the details of interacting with the API under its wrappers.

Reply:
Yeah, I think it's a fair point to discuss, and I'd generally agree our ES code is in need of a little jolt of coherence and structure. In my mind ideally the way it might work is to have ~3 layers:

- wrap the elasticsearch library for version compatibility (this new code I'm putting in `corehq.util.es`)
- wrap the version compatibility layer to present a nicer but still low-level interfaces for doing common things in elsaticsearch that includes things that are HQ-specific (sort of what's in `corehq.elastic`)
- wrap that version in a widely-used interface that is very nice to work with and fully integrated with our backends to hide low-level details (`corehq.apps.es`) which I think we're doing very well thanks to you :)

This lets us separate wrapper code that has to do with general convenience, and wrapper code that might have to be written differently depending on the elasticsearch version. Since that part will likely have to be updated every time we upgrade elasticsearch (which is at least 1 to 2, 2 to 5, 5 to 6, and 6 to 7!) I want to keep it small and well isolated from the rest of our wrapper code.

That said, while my goal is to keep it small, I may indeed try to pass _everything_ through it (mostly with very simple pass-throughs to the elasticsearch library) in order to just have a single place that, that we have control of, that everything goes through, and just to not have to call es_instance.x for this and es_interface.y for that throughout the code.

Reply:
Nice, that sounds like a good plan - I'm on board.  Thanks for laying that all out.